src/backend/distributed/metadata/dependency.c

```diff
@@ -1048,6 +1066,28 @@ GetRelationStatsSchemaDependencyList(Oid relationId)
 }
 
 
+/*
+ * CollectIndexOids implements PGIndexProcessor to create a list of all index oids
+ */
+static void
+CollectIndexOids(Form_pg_index formPgIndex, List **oids, int flags)
+{
+	*oids = lappend_oid(*oids, formPgIndex->indexrelid);
+}
+
+
+/*
+ * GetRelationIndicesDependencyList creates a list of ObjectAddressDependencies for the
+ * indexes on a given relation.
+ */
+static List *
+GetRelationIndicesDependencyList(Oid relationId)
```

Comment:
as a nice side-effect of this, functions on the indexes are also supported now:
```SQL
CREATE OR REPLACE FUNCTION lower_sql(a text)                                   RETURNS text                                                                                                         LANGUAGE sql AS $$ select lower($1); $$;
create table test(a text);
select create_distributed_table('test', 'a');
 create index i1 on test (lower_sql(a));


SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object;
...
│ (function,"{public,lower_sql}",{pg_catalog.text})                      │
...

```

Whereas on the master branch this fails:
```SQL
RROR:  function lower_sql(text) does not exist
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
CONTEXT:  while executing command on localhost:9700
```


We are already making `CREATE FUNCTION` to propagate soon via #5701, but still it is good for other objects if can be used in indexes

Reply:
I think that it is worth adding to changelog for Citus 11.

How about `Adds support for functions on index definitions` ?